File: Backend/CaffShop.Shared/DTOs/UserForRegistrationDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace CaffShop.Shared
{
  public class UserForRegistrationDto
  {
    [Required(ErrorMessage = "Name is required")]
    [MinLength(8, ErrorMessage = "Minimum 8 characters are required")]
    [MaxLength(32, ErrorMessage = "Maximum 32 characters are allowed")]
    public string Name { get; set; }

    [Required(ErrorMessage = "Username is required")]
    [MinLength(8, ErrorMessage = "Minimum 8 characters are required")]
    [MaxLength(64, ErrorMessage = "Maximum 32 characters are allowed")]
    public string Username { get; set; }

    [Required(ErrorMessage = "Password is required")]
    [MinLength(8, ErrorMessage = "Minimum 8 characters are required")]
    public string Password { get; set; }

    [Compare("Password", ErrorMessage = "Value has to match with the given password")]
    [MinLength(8, ErrorMessage = "Minimum 8 characters are required")]
    public string ConfirmPassword { get; set; }
  }
}

File: Frontend/lib/models/user_for_registration_dto.dart
class UserForRegistrationDto {
  final String name;
  final String username;
  final String password;
  final String confirmPassword;

  UserForRegistrationDto(
      {required this.name,
      required this.username,
      required this.password,
      required this.confirmPassword});

  Map<String, String> toJson() => {
        'name': name,
        'username': username,
        'password': password,
        'confirmPassword': confirmPassword
      };
}
